WeatherMonitoringSystem/Domain/Config/ConfigurationManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿
+using WeatherMonitoringSystem.Config;
+
+namespace WeatherMonitoringSystem.Domain.Config
+{
+    using Newtonsoft.Json;
+    public class ConfigurationManager
+    {
+        private static ConfigurationManager _instance;
+        private static readonly object _lock = new object();
+
+        public BotConfig RainBot { get; private set; }
+        public BotConfig SunBot { get; private set; }
+        public BotConfig SnowBot { get; private set; }
+
+        private ConfigurationManager()
+        {
+                LoadConfiguration();
+        }
+
+        public static ConfigurationManager Instance
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return _instance ??= new ConfigurationManager();
+                }
+            }
+        }
+
+        private void LoadConfiguration()
+        {
+            try
+            {
+                string relativePath = @"JSON_File\botConfig.json";
+                string configPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, relativePath);
+                if (!File.Exists(configPath))
+                {
+                    throw new FileNotFoundException("Configuration file not found.");
+                }
+
+                string json = File.ReadAllText(configPath);
+                dynamic config = JsonConvert.DeserializeObject<dynamic>(json);
+
+                RainBot = JsonConvert.DeserializeObject<BotConfig>(config["RainBot"].ToString());
+                SunBot = JsonConvert.DeserializeObject<BotConfig>(config["SunBot"].ToString());
+                SnowBot = JsonConvert.DeserializeObject<BotConfig>(config["SnowBot"].ToString());
+            }
+            catch (Exception ex)
+            {
+                throw new Exception();
```

Comment:
We should be rethrowing the same exception like this `throw;`. Throwing a new empty exception will not have the original exception message nor the stack trace. Even doing `throw ex` is wrong as well not include the stack trace.

FYI, in you specific case, you are not doing anything with the exception, so you can just remove the try-catch block and let the exception  bubble up to the caller.